pkg/gateway/middleware/app.go

```diff
@@ -20,7 +20,7 @@ func (f FindAppMiddleware) Handle(next http.Handler) http.Handler {
 			return
 		}
 
-		ctx := gatewayModel.GatewayContextFromContext(r.Context())
+		ctx := gatewayModel.Context{}
```

Comment:
I think a better approach is to make `gatewayModel.GatewayContextFromContext` lazy, i.e. return `gatewayModel.Context{}` if nil.